java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -24,18 +24,99 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 import java.util.function.Consumer;
 
+import static ai.rapids.cudf.HostColumnVector.OFFSET_SIZE;
+
 /**
  * This class represents the immutable vector of data.  This class holds
  * references to device(GPU) memory and is reference counted to know when to release it.  Call
  * close to decrement the reference count when you are done with the column, and call incRefCount
  * to increment the reference count.
  */
-public final class ColumnVector implements AutoCloseable, BinaryOperable {
+public final class ColumnVector implements AutoCloseable, BinaryOperable, ColumnViewPointerAccess<BaseDeviceMemoryBuffer> {
   private static final Logger log = LoggerFactory.getLogger(ColumnVector.class);
 
+  public class DeviceColumnViewPointerAccess implements ColumnViewPointerAccess<BaseDeviceMemoryBuffer> {
+
+    protected long viewHandle;
+
+    public DeviceColumnViewPointerAccess(long viewHandle) {
+      this.viewHandle = viewHandle;
+    }
+
+    @Override
+    public long getColumnView() {
+      return viewHandle;
+    }
+
+    @Override
+    public ColumnViewPointerAccess<BaseDeviceMemoryBuffer> getChildColumnView(int childIndex) {
```

Comment:
This should be called `getChilColumnViewPointerAccess` exactly what it returns

Reply:
:( very long name :( but happy to make the change.

Reply:
The more I think about it - it is not a simple getter on the class so maybe we can keep this one as is , what do u think @razajafri ?

Reply:
Will it make sense to change the name of the class to be ColumnViewAccess? That will make it shorter and avoid the ambiguity

Reply:
sounds good. @revans2 I know your related PRs will be impacted by this public api change, let me know if that's cool with u and I will psuh a change here soon.

Reply:
Made the change for now. @razajafri please review :)